src/osd/PeeringState.cc

```diff
@@ -73,22 +63,24 @@ void BufferedRecoveryMessages::send_info(
   const pg_info_t &info)
 {
   if (require_osd_release >= ceph_release_t::octopus) {
-    message_map[to].push_back(
+    send_osd_message(
+      to,
       make_message<MOSDPGInfo2>(
 	to_spgid,
 	info,
 	cur_epoch,
 	min_epoch)
       );
   } else {
-    auto m = make_message<MOSDPGInfo>(cur_epoch);
-    m->pg_list.push_back(
-      pg_notify_t(
-	to_spgid.shard,
-	info.pgid.shard,
-	min_epoch, cur_epoch,
-	info, PastIntervals()));
-    message_map[to].push_back(m);
+    send_osd_message(
+      to,
+      make_message<MOSDPGInfo>(
```

Comment:
```cpp
class MOSDPGInfo : public Message {
  // ...

public:
  using pg_list_t = std::vector<pg_notify_t>;
  pg_list_t pg_list;

  // ...
  MOSDPGInfo(epoch_t mv)
    : MOSDPGInfo(mv, {})
  {}
  MOSDPGInfo(epoch_t mv, pg_list_t&& l)
    : Message{MSG_OSD_PG_INFO, HEAD_VERSION, COMPAT_VERSION},
      epoch{mv},
      pg_list{std::move(l)}
  {
    set_priority(CEPH_MSG_PRIO_HIGH);
  }
```
OK, looks like this is change pure 1-to-1 clean-up. Before we were just using delegating constructor.